File: 월간 코드 챌린지 시즌2/2개 이하로 다른 비트.py
numbers = [2,7]

def f(x):
    if x % 2 == 0:
        return x+1
    
    else:
        y = '0' + bin(x)[2:]
        idx = y.rfind('0')
        y = list(y)
        y[idx] = '1'
        y[idx+1] = '0'
                
        return int(''.join(y), 2)

def solution(numbers):
    
    answer = [f(i) for i in numbers]
        
    return answer

print(solution(numbers))

# def solution(numbers):
#     answer = []
#     for number in numbers:
        
#         bin_number = list('0' + bin(number)[2:])
#         idx = ''.join(bin_number).rfind('0')
#         bin_number[idx] = '1'
        
#         #홀수일땐 idx의 다음 숫자를 0으로 바꿔준다.
#         if number%2!=0:
#             bin_number[idx+1]='0'
#         answer.append(int("".join(bin_number),2))
#     return answer

File: 월간 코드 챌린지 시즌2/2개 이하로 다른 비트.js
function solution(numbers) {
  var answer = [];
  for (let i = 0; i < numbers.length; i++) {
    let start = numbers[i] + 1;
    while (true) {
      let num = numbers[i].toString(2);
      const a = start.toString(2);
      if (a.length > num.length) {
        const l = a.length - num.length;
        num = "0".repeat(l) + num;
      }
      result = 0;
      for (let j = 0; j < a.length; j++) {
        if (a[j] !== num[j]) {
          result += 1;
        }
      }
      if (result < 3) {
        answer.push(start);
        break;
      }
      start += 1;
    }
  }
  return answer;
}

numbers = [2, 7];
console.log(solution(numbers));
